apps/server/src/services/message-service/MessageService.ts

```diff
@@ -36,9 +39,42 @@ class MessageService {
       visible: false,
     };
 
+    this.externalMessage = {
+      text: '',
+      visible: false,
+    };
+
     this.onAir = false;
   }
 
+  private _OnTimeOut() {
+    eventStore.batchSet(this.getAll())
+    this.reateLimit = null
+  }
+
+  /**
+   * @description sets message on stage timer screen
+   */
+  setExternalText(payload: string) {
+    if (this.externalMessage.text !== payload) {
+      this.externalMessage.text = payload;
+      if (!this.reateLimit) {
+        eventStore.set('externalMessage', this.externalMessage);
+        this.reateLimit = setTimeout(this._OnTimeOut.bind(this), 1000);
```

Comment:
@cpvalente what do you think of this as rate limiting?

Reply:
hmm the UI text fields don't really work with this method as they send an update on each key press

Reply:
Few points:
- we want to always have the latest information in the socket, with this logic, we may not send the data if we are in timeout period
- it seems that the rate limiting needs to be a bit less ambitious, maybe around 64ms? we need to test with the UI

Take a look into debouncing code, pretty similar to what you have, but should ensure that we always send the latest call